Clients/GUI/Pages/InstrumentInfo.razor

```diff
@@ -0,0 +1,469 @@
+            {
+                tempDataSet.RemoveAt(0);
+            }
+            lineConfig.Data.Datasets.Add(tempDataSet);
+
+            lineChartJs.Update();
+        }
+    }
+
+    private void AddCandles(IEnumerable<Candle> candles)
+    {
+        tempDataSet.AddRange(candles.ToList().ConvertAll(candle =>
+            new TimeTuple<decimal>(new Moment(candle.Time), candle.Close)));
+
+        lineConfig.Data.Datasets.Add(tempDataSet);
+
+        lastTime = candles.Last().Time;
+    }
+
+    private void OnError()
+    {
+        lineConfig.Options.Title.Text = "Instrument not found";
+
+        lineChartJs.Update();
+    }
+
+    protected override void OnInitialized()
+    {
+        SetChartConfig();
+    }
+
+    private void SetChartConfig()
+    {
+        lineConfig = new LineConfig
+        {
+            Options = new LineOptions
+            {
+                Responsive = true,
+                Title = new OptionsTitle
+                {
+                    Display = true,
+                    Text = ""
+                },
+                Tooltips = new Tooltips
+                {
+                    Mode = InteractionMode.Nearest,
+                    Intersect = false
+                },
+                Scales = new Scales
+                {
+                    xAxes = new List<CartesianAxis>
+        {
+            new TimeAxis
+            {
+                Distribution = TimeDistribution.Linear,
+                Ticks = new TimeTicks
+                {
+                    Source = TickSource.Data
+                },
+                Time = new TimeOptions
+                {
+                    Unit = TimeMeasurement.Minute,
+                    Round = TimeMeasurement.Minute,
+                    TooltipFormat = "DD.MM.YYYY HH:mm",
+                    DisplayFormats = TimeDisplayFormats.DE_CH
+                }
+            }
+        }
+                },
+                Hover = new LineOptionsHover
+                {
+                    Intersect = true,
+                    Mode = InteractionMode.Y
+                },
+                Legend =
+    {
+        Display = false
+    }
+            }
+        };
+
+        tempDataSet = new LineDataset<TimeTuple<decimal>>
+        {
+            BorderColor = ColorUtil.FromDrawingColor(Color.CornflowerBlue),
+            Fill = false,
+            BorderWidth = 2,
+            PointRadius = 3,
+            PointBorderWidth = 1,
+            SteppedLine = SteppedLine.False
+        };
+    }
+
+    private void UpdateBalanceVariable()
+    {
+        currentBalance = currency switch
+        {
+            Currency.Rub => userBalance.BalanceInRub,
+            Currency.Usd => userBalance.BalanceInUsd,
+            Currency.Eur => userBalance.BalanceInEur
+        };
+    }
+
+    private void UpdateBalanceInCorrectCurrency()
+    {
+        switch (currency)
+        {
+            case Currency.Rub:
+                userBalance.BalanceInRub += cash;
```

Comment:
Actually connection between "update" and "+=" is not obvious (at least for me), but why not

Reply:
Suggestions?

Reply:
> Suggestions?

"Increase balance" with += or "update balance" with =, idk

Reply:
fixed